Reintroduce authorization documentation (#3673)
apollographql/router: .changesets/maint_geal_authorization_analytics.md

### Add a metric tracking authorization usage ([PR #3660](https://github.com/apollographql/router/pull/3660))

The new metric is a counter called `apollo.router.operations.authorization` and contains the following boolean attributes:

> - filtered: some fields were filtered from the query
> - requires_scopes: the query uses fields or types tagged with the `@requiresScopes` directive
> - authenticated: the query uses fields or types tagged with the `@authenticated` directive

Reviewer: ```suggestion
- `filtered`: the query has one or more filtered fields 
- `requires_scopes`: the query uses fields or types tagged with the `@requiresScopes` directive
- `authenticated`: the query uses fields or types tagged with the `@authenticated` directive
```
